mne/decoding/transformer.py

```diff
@@ -715,3 +715,132 @@ def transform(self, X):
+        forward-backward filtering (via filtfilt).
+    iir_params : dict | None
+        Dictionary of parameters to use for IIR filtering.
+        See mne.filter.construct_iir_filter for details. If iir_params
+        is None and method="iir", 4th order Butterworth will be used.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+        Defaults to self.verbose.
+    """
+    def __init__(self, l_freq=None, h_freq=None, sfreq=1.0,
+                 filter_length='10s', l_trans_bandwidth=None,
+                 h_trans_bandwidth=None, n_jobs=1, method='fft',
+                 iir_params=None, verbose=None):
+        self.l_freq = l_freq
+        self.h_freq = h_freq
+        self.filter_length = filter_length
+        self.l_trans_bandwidth = l_trans_bandwidth
+        self.h_trans_bandwidth = h_trans_bandwidth
+        self.n_jobs = n_jobs
+        self.method = method
+        self.iir_params = iir_params
+
+        if not isinstance(sfreq, (int, float)):
+            raise ValueError("sfreq must be of type int or float, got %s "
+                             "instead" %type(sfreq))
+        self.sfreq = float(sfreq)
+        if self.l_freq == 0:
+            self.l_freq = None
+        if self.h_freq is not None and self.h_freq > (self.sfreq / 2.):
+            self.h_freq = None
+        if self.l_freq is not None and not isinstance(self.l_freq, float):
+            self.l_freq = float(self.l_freq)
+        if self.h_freq is not None and not isinstance(self.h_freq, float):
+            self.h_freq = float(self.h_freq)
+
+
+    def fit(self, X, y=None):
+        """Does nothing. For scikit-learn compatibility purposes.
+
+        Parameters
+        ----------
+        X : array, shape (n_epochs, n_channels, n_times)
+            The data.
+        y : array, shape (n_epochs,) | None
+            The label for each epoch.
+
+        Returns
+        -------
+        self : instance of Filterer
+            Returns the modified instance.
+        """
+        if not isinstance(X, np.ndarray):
+            raise ValueError("X should be of type ndarray (got %s)."
+                             % type(X))
+
+        return self
+
+    def transform(self, X, y=None):
+        """Filters data along the third axis.
+
+        Parameters
+        ----------
+        X : array, shape (n_epochs, n_channels, n_times)
+            The data.
+        y : None | array, shape (n_epochs,)
+            The label for each epoch.
+            If None not used. Defaults to None.
+
+        Returns
+        -------
+        X : array, shape (n_epochs, n_channels, n_times)
+            The data after filtering.
+        """
+        if not isinstance(X, np.ndarray):
+            raise ValueError("epochs_data should be of type ndarray (got %s)."
+                             % type(X))
+
+        X = np.atleast_3d(X)
+        X = filter_data(X, self.sfreq, self.l_freq, self.h_freq,
```

Comment:
Maybe concatenate the epochs with the channels , as long as they are
filtered independlty that should be fine.

Le 30 juil. 2016 13:14, "Asish Panda" <notifications@github.com> a écrit :

> In mne/decoding/transformer.py
> <https://github.com/mne-tools/mne-python/pull/3472#discussion_r72891004>:
>
> > +            The data.
> > +        y : None | array, shape (n_epochs,)
> > +            The label for each epoch.
> > +            If None not used. Defaults to None.
> > +
> > +        Returns
> > +        -------
> > +        X : array, shape (n_epochs, n_channels, n_times)
> > +            The data after filtering.
> > +        """
> > +        if not isinstance(X, np.ndarray):
> > +            raise ValueError("epochs_data should be of type ndarray (got %s)."
> > +                             % type(X))
> > +
> > +        X = np.atleast_3d(X)
> > +        X = filter_data(X, self.sfreq, self.l_freq, self.h_freq,
>
> @kingjr <https://github.com/kingjr> this fails as filter_data takes only
> two 2D data. However looking at the code I don't really see why it has been
> restriced to 2D data. filer_data
> <https://github.com/mne-tools/mne-python/blob/master/mne/filter.py#L667>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/3472/files/2c25975499c15dc755a2b67d6e79ba0cc8c292cd#r72891004>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEp7DKYkWw-Y2SuNWkQbLfx3lT1owS8rks5qa4aQgaJpZM4JYh5i>
> .
>